ckanext/dsaudit/actions.py

```diff
@@ -25,11 +25,19 @@ def fresh_context(context):
     )
 
 
+def _is_system_user(context):
+    site_user = get_action("get_site_user")({"ignore_auth": True}, {})
+    if site_user["name"] == context.get('user') or not context.get('user'):
+        return True
+    return False
+
+
 @chained_action
 def datastore_create(original_action, context, data_dict):
     rval = original_action(context, data_dict)
     res = context['model'].Resource.get(rval['resource_id'])
-    if res.url_type not in h.datastore_rw_resource_url_types():
+    # we want to record activity for `upload` type Data Dcitionary saving
+    if (res.url_type != 'upload' and res.url_type not in h.datastore_rw_resource_url_types()) or _is_system_user(context):
```

Comment:
I think this can be simplified
```suggestion
    if res.url_type not in h.datastore_rw_resource_url_types() or _is_system_user(context):
```
On our site only uploaded resources will have a data dictionary and on other sites they may use datapusher/xloader on remote urls and still want this history